revops/revops.js

```diff
@@ -0,0 +1,48 @@
+async function populateServices() {
+  const request = new Request("services.json");
+
+  const response = await fetch(request);
+  const services = await response.json();
+
+  const servicesList = document.createElement("div");
+  servicesList.classList.add("collection-list", "w-dyn-items");
+
+  services.forEach((service) => {
+    const serviceItem = document.createElement("div");
+    serviceItem.classList.add("collection-item", "w-dyn-item");
+
+    const serviceItemLink = document.createElement("a");
+    serviceItemLink.href = service.Slug;
+    serviceItemLink.classList.add("link-block", "w-inline-block");
+
+    const serviceItemContainer = document.createElement("div");
+    serviceItemContainer.classList.add("div-block");
+
+    const serviceItemIcon = document.createElement("img");
+    serviceItemIcon.src = service.Icon;
+    serviceItemIcon.classList.add("image");
+
+    const serviceItemTitle = document.createElement("h3");
+    serviceItemTitle.innerHTML = service.Name;
+    serviceItemTitle.classList.add("post-preview-title");
+
+    const serviceItemSummary = document.createElement("p");
+    serviceItemSummary.innerHTML = service["Short description"];
+    serviceItemSummary.classList.add("post-summary");
```

Comment:
Instead of building elements like this, I usually create a top-level element and create the rest of the structure with `innerHTML`:

```js
const servicesList = document.createElement("div");
servicesList.classList.add("collection-list", "w-dyn-items");
servicesList.innerHTML = services.map((service) => `
  <div class="collection-item w-dyn-item">
    <a href="${service.Slug}" class="link-block w-inline-block">
      <div class="div-block">
        <img src="${service.Icon}">
        <h3 class="post-preview-title">${service.Name}</h3>
        <p class="post-summary">${service["Short description"]}</p>
      </div>
    </a>
  </div>
`).join("");
```

This leans into the structure of HTML and keeps us closer to a single way of defining HTML.

Reply:
Yes, I agree building HTML the way I did it here is inefficient, I have used different method for the 'latest blog posts' list, utilizing <template>, but what you are suggesting might be even better

Reply:
Fixed